src/bin/add/fetch.rs

```diff
@@ -247,6 +232,29 @@ quick_error! {
     }
 }
 
+macro_rules! get_crate_name_from_repository {
```

Comment:
Nice touch with the macro, but… can we make that a function with a signature like `(repo: &str, matcher: &Regex, url_template: Fn(user: &str, repo: &str) -> String)`?